From d80e83329434889c8f8e9429870116e409b0110b Mon Sep 17 00:00:00 2001
From: sultanxda <sultanxda@gmail.com>
Date: Thu, 8 Mar 2018 23:35:48 +0100
Subject: [PATCH 37/57] camera: Fix high battery drain after using flashlight

After turning off the flashlight via the tile, the camera is kept open,
preventing the device from entering deep sleep (and destroying battery
life).

Properly close the camera after turning off the flashlight in order to fix
the high battery drain.
---
 services/camera/libcameraservice/CameraFlashlight.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/camera/libcameraservice/CameraFlashlight.cpp b/services/camera/libcameraservice/CameraFlashlight.cpp
index 3979056c5..dafda589a 100644
--- a/services/camera/libcameraservice/CameraFlashlight.cpp
+++ b/services/camera/libcameraservice/CameraFlashlight.cpp
@@ -571,7 +571,7 @@ status_t CameraHardwareInterfaceFlashControl::disconnectCameraDevice() {
     }
     mDevice->setPreviewWindow(NULL);
     mDevice->release();
-    mDevice = NULL;
+    mDevice.clear();
 
     return OK;
 }
-- 
2.11.0

